### PR TITLE
re-Implement isMember for token based DAOs by calling the contract

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- re-Implement isMember for token based DAOs by calling the contract (protected route logic)

Task: [APP-2629](https://aragonassociation.atlassian.net/browse/APP-2629)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2629]: https://aragonassociation.atlassian.net/browse/APP-2629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/aragon-sdk/queries/use-is-member.ts [hunk 1]
```diff
@@ -39,16 +26,100 @@ export const useIsMember = (
 ) => {
   const {network} = useNetwork();
   const client = usePluginClient(params.pluginType);
+  const {api: provider} = useProviders();
+  const fetchVotingPower = useVotingPowerAsync();
+  const {data: votingSettings, isLoading: settingsAreLoading} =
```
Comment (Fabricevladimir) [inline src/services/aragon-sdk/queries/use-is-member.ts:40]: Gimme a space for readability please. 
Also, considering this query is only for the token based plugin, I would only enable it if `params.pluginType` is the token based plugin.

File: src/services/aragon-sdk/queries/use-is-member.ts [hunk 2]
```diff
@@ -39,16 +26,100 @@ export const useIsMember = (
...
+    useVotingSettings({
+      pluginAddress: params.pluginAddress,
+      pluginType: params.pluginType,
+    });
+  const {
```
Comment (Fabricevladimir) [inline src/services/aragon-sdk/queries/use-is-member.ts:50]: Gimme another space please lol

**Review by Fabricevladimir — COMMENTED** [review]
Thanks, Sepehr! It would be nice if you could extract the fetching functions outside of the hook, but it's not a big deal if you can't.

**Comment by sepehr2github** [comment]
> Thanks, Sepehr! It would be nice if you could extract the fetching functions outside of the hook, but it's not a big deal if you can't.

I thought about it as well but It should be tackled on a different ticket

File: src/services/aragon-sdk/queries/use-is-member.ts [hunk 3]
```diff
@@ -7,48 +7,130 @@ import {PluginTypes, usePluginClient} from 'hooks/usePluginClient';
...
+  const {api: provider} = useProviders();
+  const fetchVotingPower = useVotingPowerAsync();
+
+  // fetch voting settings
+  const {data: votingSettings, isLoading: settingsAreLoading} =
+    useVotingSettings({
+      pluginAddress: params.pluginAddress,
+      pluginType: 'token-voting.plugin.dao.eth',
+    });
+
```
Comment (Fabricevladimir) [inline src/services/aragon-sdk/queries/use-is-member.ts]:
```suggestion
 const {data: votingSettings, isLoading: settingsAreLoading} =
    useVotingSettings(
      {
        pluginAddress: params.pluginAddress,
        pluginType: params.pluginType,
      },
      {enabled: params.pluginType === 'token-voting.plugin.dao.eth'}
    );

```

File: src/services/aragon-sdk/queries/use-is-member.ts [hunk 4]
```diff
@@ -7,48 +7,130 @@ import {PluginTypes, usePluginClient} from 'hooks/usePluginClient';
...
+  // fetch dao members
+  const {
+    data: {daoToken, filteredMembers},
+    isLoading: membersAreLoading,
+  } = useDaoMembers(params.pluginAddress, 'token-voting.plugin.dao.eth', {
+    searchTerm: params.address as string,
+    page: 0,
+  });
```
Comment (Fabricevladimir) [inline src/services/aragon-sdk/queries/use-is-member.ts]:
```suggestion
  const {
    data: {daoToken, filteredMembers},
    isLoading: membersAreLoading,
  } = useDaoMembers(params.pluginAddress, params.pluginType, {
    searchTerm: params.address,
    page: 0,
    enabled: params.pluginType === 'token-voting.plugin.dao.eth',
  });
```

**Review by Fabricevladimir — COMMENTED** [review]
Added a couple of suggestions. Also, can you fix the overflowing of the errors on the verification card on mobile?
Here's a proposal where this happens for reference: http://localhost:3000/#/daos/goerli/again.dao.eth/governance/proposals/0xa03bf96689b988cc9a9489ac5dd7aba6900064b7_0x4

**Review by Fabricevladimir — APPROVED** [review]

**Pull request merged**